docs/architecture/Data_caching_lineage.md

```diff
@@ -0,0 +1,297 @@
+
+1. PythonScriptStep with custom code on top of the Azure Storage SDK or [AzCopy](https://docs.microsoft.com/en-us/azure/storage/common/storage-ref-azcopy-copy).
+1. Durable Azure Functions with custom code on top of the Azure Storage SDK or AzCopy.
+1. [DataTransferStep](https://docs.microsoft.com/en-us/python/api/azureml-pipeline-steps/azureml.pipeline.steps.data_transfer_step.datatransferstep?view=azure-ml-py), a specialized step in Azure Machine Learning, utilizes Azure Data Factory (Copy Activity).
+1. PythonScriptStep directly initiating an Azure Data Factory pipeline with [Copy Activity](https://docs.microsoft.com/en-us/azure/data-factory/copy-activity-overview).  
+
+ Option | Pros | Cons |
+| -------- | --- | ----------- |
+| PythonScriptStep with AzCopy | Fast and efficient, provides required features | Storage credentials must be exposed to the step (directly, via Key Vault, or to AML compute's identity) |<!-- markdownlint-disable MD013 -->
+| Durable Functions | Fast and efficient, provides almost all required features, no need to expose storage credentials to pipelines | Significantly increases run orchestration complexity, adds manageability overhead, no access to AML run output |<!-- markdownlint-disable MD013 -->
+| DataTransferStep | Lowest manageability overhead, storage credentials are managed within AML Datastores | Doesn't support incremental copying |
+| Azure Data Factory | no need to expose storage credentials to pipelines | Same as above + manageability overhead |<!-- markdownlint-disable MD013 -->
+
+Option "PythonScriptStep with AzCopy" is a clear winner by the following reasons:
+
+1. It has greater flexibility and AzCopy provides a highly efficient data transfer mechanism with maximum performance.
+1. It runs within AML pipelines, and has access to all pipeline's and run's subsystems.
+1. Storage credentials exposure can be mitigated (see below).
+
+### Managing Storage Credentials in AML Pipelines
+
+When a certain code component running within an AML pipeline requires direct access to Azure Storage, there are multiple ways to provide it:
+
+1. Pass Account Key or SAS-token as AML Pipeline Parameters (**not recommended**).
+1. Store storage access credentials in [AML workspace's Key Vault](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-use-secrets-in-runs)
+1. Isolate code that requires direct storage access to a separate PythonScriptStep (ParallelRunStep) running on an AML compute cluster with a [managed identity](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview). Such identity must be given appropriate access to the storage resources. This option should be used once Managed Identity support in azure ML reaches General Availability status.
```

Comment:
Is there a way to provide a pointer/link to check on the status of the Managed Identity support in AML?